src/scripts/common/renderTopMenu.js

```diff
@@ -0,0 +1,51 @@
+import { authentication } from './authentication';
+import { ElementHelper } from './ElementHelper';
+
+// Logging out for eventListener
+const onPressLogout = () => (event) => {
+    event.preventDefault();
+    authentication.signout();
+}
+
+export const renderTopMenu = (user) => {
+    // Pick top menu elements
+    const hamburger = document.getElementById('menuButton');
+    const ul = document.querySelector('.menubar');
+    const liProfile = document.createElement('li');
+    const liSettings = document.createElement('li');
+    const liLogout = document.createElement('li');
+    const liSignin = document.createElement('li');
+    const liSignup = document.createElement('li');
+
+    // Toggle hamburger menu on and off (hidden, not hidden) when button clicked.
+    // When menu is shown, prevent scrolling.
+    hamburger.addEventListener('click', (event) => {
+        event.preventDefault();
+        ul.classList.toggle('hidden');
+
+        if (!ul.classList.contains('hidden')) {
+            document.body.style.overflowY = 'hidden';
+        } else {
+            document.body.style.overflowY = '';
+        }
+    });
+
+    // If user if logged in, show buttons for profile, settings and logout.
+    // If not, show buttons for sign in and sign up.
+    if (user) {
+        ul.appendChild(liProfile);
```

Comment:
You might be able to refactor this a little bit to make it even shorter and more straight-forward with the new change to the `ElementHelper` class I mentioned in a previous comment. For example here you could create the `li` element and append it to the `ul` in just one line
```js
const liProfile = ElementHelper.create('li').setParent(ul);
```
and then add the content of the `li` even simpler:
```js
ElementHelper.create('a').setHref(`/profile?id=${user.id}`).setText('Profile').setParent(liProfile);
```
This way you need to use less `createElement` and `appendChild` mess.